pkg/verifier/verify.go

```diff
@@ -5,50 +5,78 @@ package verifier
 
 import (
 	"context"
+	"time"
 
 	"github.com/thanos-io/thanos/pkg/block"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/oklog/ulid"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/thanos-io/thanos/pkg/objstore"
 )
 
+type verifierMetrics struct {
+	blocksMarkedForDeletion prometheus.Counter
+}
+
+func newVerifierMetrics(reg prometheus.Registerer) *verifierMetrics {
+	var m verifierMetrics
+
+	m.blocksMarkedForDeletion = prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
`promauto.With(reg).NewCounter` is available here :p Having this metric here is OK since the `issueFn` functions do not use our syncer underneath.

Reply:
Changed the counter construction, using promauto now.